main.py

```diff
@@ -470,8 +470,74 @@ def casenum_register():
 @app.route('/form/submit', methods=['POST'])
 def casenum_updaterecord():
 
-    caseNum = "2019-0106-6370"
-    record = get_dynamodb_record(caseNum)
+    #
+    # Response Format
+    #
+    jsonResponse = {
+        'status': 'error',
+        'message': '',
+        'http_code': 403,
+        'case_number': ''
+    }
+
+    #
+    # Keep generating a case number until a new record is created.
+    #
+
+    while True:
+        caseNum = generate_casenum() # Generate case num.
+        print("casenum_register() Case Number: " + caseNum)
+        record = get_dynamodb_record(caseNum) # Record is 'None' if not found.
+
+        if(record == None):
+            print("casenum_register() Record not found, creating new...")
+            caseNumResp, resp = create_dynamodb_record(inputJson='{"type": "new_form_submission_placeholder"}',case_number=caseNum)
```

Comment:
I feel like it might make more sense to update `create_dynamodb_record()` to have the logic to make sure we have a unique case number. I may be missing something though. Is there a reason that we have the check here instead?

Reply:
The case numbers are indexed, I believe that if I create a 2nd entry with the same case number it will likely fail to create the record on Dynamo.

So the loop goes like this:

1. Generate random number. `via generate_casenum()`
2. Check if that number can be found `via get_dynamodb_record()`
3. If not, then create it `via create_dynamodb_record()`. 
    Else repeat steps.

Reply:
@briaguya Well, dang. It looks like dynamodb is not like relational dbs. I tried inserting an existing value and it just updates the record.